bytes/bytes/raw/file_raw_repository.py

```diff
@@ -62,3 +72,63 @@ def _raw_file_path(self, raw_id: UUID, boefje_meta: BoefjeMeta) -> Path:
 
     def _index(self, raw_id: UUID) -> str:
         return str(raw_id)[: self.UUID_INDEX]
+
+
+class S3RawRepository(RawRepository):
+    def __init__(
+        self,
+        file_middleware: FileMiddleware,
+        bucket_per_org: bool,
+        s3_bucket_prefix: str = "OpenKAT-",
+        s3_bucket_name: str = "OpenKAT",
+    ) -> None:
+        self._file_middleware = file_middleware
+        self.bucket_per_org = bucket_per_org
+        self.s3_bucket_prefix = s3_bucket_prefix
+        self.s3_bucket_name = s3_bucket_name
+
+    @cached_property
+    def _s3resource(self):
+        session = BotoSession()
+        return session.resource("s3")
+
+    def get_or_create_bucket(self, organization: str):
+        # Create a bucket, and if it exists already return that instead
+        bucket_name = self.s3_bucket_name
+        if self.bucket_per_org:
+            bucket_name = f"{self.s3_bucket_prefix}{organization}"
+            try:
```

Comment:
If we expect that the common case is that the bucket exists (because we create it only once), would it be possible to flip the logic here and do something like this?

```python3
try:
  return self._s3resource.Bucket(name=bucket_name)
except BucketDoesNotExist:
  bucket = self._s3resource.create_bucket(Bucket=bucket_name)
  bucket.wait_until_exists()
  return bucket
```

Then at least we have 1 error and N happy flows in general instead of the other way around, but not sure if that Bucket() call fails if no bucket exists, meaning we make only 1 call instead of 2 calls always.

Reply:
After some testing, I found out that no exception gets thrown when trying to create a bucket that already exists (according to the documentation it does throw an exception, but that part is outdated then I assume).

So now instead, we can create our bucket every time it is needed and it will give us the old bucket.

```python
try:
    bucket = self._s3resource.create_bucket(Bucket=bucket_name)
    bucket.wait_until_exists()
    return bucket
except bucket.meta.client.exceptions.ClientError as error:
    logger.error("Something went wrong)
    raise error 
```

An exception can still occur, but is not expected.
Implemented in [264733a](https://github.com/minvws/nl-kat-coordination/pull/3505/commits/264733a1d127df640eaabbcad9c9b1b76cfe5538)

Reply:
I'm not sure if the solution is reliable but one suggestion here is to use the creation_date field to check if we need to create the bucket still: https://stackoverflow.com/questions/26871884/how-can-i-easily-determine-if-a-boto-3-s3-bucket-resource-exists. One instance where we might still have significant overhead is when creating multiple raw files at once under higher workloads. It still feels a bit like the creating-a-connection-per-database-operation anti-pattern to create the bucket redundantly on every request, but perhaps most users just use the one-bucket setup